### ISSUE TITLE
Apple Pay Button Responsive

### ISSUE THREAD

**Issue by SaadmLMC** [flutter-stripe/flutter_stripe#1287]
Describe the bug
I followed the apple pay integration. Generated certificates from Apple Developers account and Stripe dashboard and got it approved. Also made sure to match the Merchant Id everywhere. 

Still Apple Pay button is unresponsive.

 To Reproduce
1. Entering number 4242424424242 into the card field.
2. Tapping the confirm button.
3. The stripe payment sheet disappears

Expected behavior
Initiate Apple Pay on the Mobile Pay

**Comment by remonh87** [comment]
Can you give me a reproducible example of what you are doing? 

You mentioned the `card field` but it is not used for Apple Pay.  You can either use the paymentsheet (recommended way) or use our[ Apple Pay flow](https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/apple_pay_screen.dart).

**Comment by SaadmLMC** [comment]
> Can you give me a reproducible example of what you are doing?
> 
> You mentioned the `card field` but it is not used for Apple Pay. You can either use the paymentsheet (recommended way) or use our[ Apple Pay flow](https://github.com/flutter-stripe/flutter_stripe/blob/main/example/lib/screens/wallets/apple_pay_screen.dart).

I'm sorry for the typo, there's no card number input, and sure I can share the snippet of the code

**Comment by SaadmLMC** [comment]
```
  void payStripe (String amount) async {
    double new_amount = double.parse(amount);
    Map<String, String> headers = {
      HttpHeaders.authorizationHeader: 'Bearer ${stripPaymentKey.validate()}',
      HttpHeaders.contentTypeHeader: 'application/x-www-form-urlencoded',
    };

    var request = http.Request('POST', Uri.parse(stripeURL));

    request.bodyFields = {
      'amount': '${(new_amount * 100).toInt()}',
      'currency': "${appStore.currencyName.toUpperCase()}",
    };

    log(request.bodyFields);
    request.headers.addAll(headers);

    log(request);

    appStore.setLoading(true);

    await request.send().then((value) {
      appStore.setLoading(false);
      http.Response.fromStream(value).then((response) async {
        if (response.statusCode == 200) {
          var res = StripePayModel.fromJson(await handleResponse(response));

          SetupPaymentSheetParameters setupPaymentSheetParameters = SetupPaymentSheetParameters(
            paymentIntentClientSecret: res.clientSecret.validate(),
            style: ThemeMode.light,
            appearance: PaymentSheetAppearance(colors: PaymentSheetAppearanceColors(primary: primaryColor)),
             applePay:  PaymentSheetApplePay(merchantCountryCode: appStore.currencyName.toUpperCase()),
            googlePay: PaymentSheetGooglePay(merchantCountryCode: appStore.currencyName.toUpperCase(), testEnv: true),
            merchantDisplayName: mAppName,
            customerId: appStore.userId.toString(),
            //customerEphemeralKeySecret: res.clientSecret.validate(),
            //setupIntentClientSecret: res.clientSecret.validate(),
          );

          await Stripe.instance.initPaymentSheet(paymentSheetParameters: setupPaymentSheetParameters).then((value) async {
            await Stripe.instance.presentPaymentSheet().then((value) async {
              saveBookingData();
            });
          }).catchError((e) {
            log("presentPaymentSheet ${e.toString()}");
          });
        }
      }).catchError((e) {
        appStore.setLoading(false);
        toast(e.toString(), print: true);
      });
    }).catchError((e) {
      appStore.setLoading(false);
      toast(e.toString(), print: true);
    });
  }
```

**Comment by nihaalnz** [comment]
@remonh87 
I am having this issue as well, very similar to the code, the documents seems to be outdated about using apple pay

**Comment by iridiscenthummingbird** [comment]
Have the same issue

**Comment by iridiscenthummingbird** [comment]
UPD

**Comment by iridiscenthummingbird** [comment]
UPD
I used correct country code and it works

**Comment by remonh87** [comment]
I the weird thing is that in my project setup params works flawlessly with Apple Pay.  But I saw two things: 

- Did you also set the merchant identifier like https://github.com/flutter-stripe/flutter_stripe/blob/bf9b83e942cee5b52db76144e4c7748f34da158a/example/lib/main.dart#L10 ? 

- For merchantcountrycode you do: `merchantCountryCode: appStore.currencyName.toUpperCase())` . This code looks like you put something like `USD` in that value while you should put an ISO 3166 for this value (like 'DE' or 'US'.

**Comment by SaadmLMC** [comment]
> I the weird thing is that in my project setup params works flawlessly with Apple Pay. But I saw two things:
> 
> * Did you also set the merchant identifier like https://github.com/flutter-stripe/flutter_stripe/blob/bf9b83e942cee5b52db76144e4c7748f34da158a/example/lib/main.dart#L10
>    ?
> * For merchantcountrycode you do: `merchantCountryCode: appStore.currencyName.toUpperCase())` . This code looks like you put something like `USD` in that value while you should put an ISO 3166 for this value (like 'DE' or 'US'.
> 
> I also would recommend to contact Stripe support because we have little control over the widget itself.

I checked, my merchant identifier is fine and approved. And the merchant country code is getting the right value in proper format - ISO 3166 Alpha 2

However, I'll try to downgrade and try my luck with one of the old versions. I remember Apple Pay working for me couple of months pack on the previous version.

**Comment by remonh87** [comment]
Any success? I think it should be a simple misconfiguration of the certificate. I am using Apple Pay for two different production projects. It is a bit tricky however and you don't get proper feedback. However next release we hopefully provide better error messages

**Comment by wizard11** [comment]
I was also facing the same issue. But then I tried changing the country code from UK to US as pointed in this thread above. It worked for me as well. Then I went back and checked the Apple's and Stripe's documentation. The Apple's doc says `Consult with your payment service provider (PSP) to determine the appropriate country code.` and the Stripe [doc](https://stripe.com/docs/apple-pay) says nothing about this country code and only shows the example where it uses the "US" as merchantCountryCode. I tried the "US" merchantCountryCode with different currency codes - "usd", "gbp" & "eur" and all the amounts work fine with. So my conclusion is since Stripe is US based, we need to always use "US" as merchant country code and then we are ok to use any currency code

**Comment by remonh87** [comment]
@wizard11  that is incorrect. You need to use the country of the merchant that you registered  with on Apple (this must match). I implemented apple pay with this library for merchants in Germany and also Sweden without issues.

**Comment by remonh87** [comment]
closing it for now feel free to reopen it or create a new issue in case any problem persist
